asf_heat_pump_suitability/analysis/hn_zones/plot_comparison_of_hn_zones.py

```diff
@@ -0,0 +1,225 @@
+)
+LIVERPOOL_HP_SUITABILITY_PARQUET = os.path.join(
+    INPUT_DIR, "liverpool_hp_suitability_scores_with_desnz.parquet"
+)
+LSOA_SHP_PATH = "s3://asf-heat-pump-suitability/source_data/Lower_layer_Super_Output_Areas_2021_EW_BFE_V9_-9107090204806789093/LSOA_2021_EW_BFE_V9.shp"
+LIVERPOOL_GPKG_PATH = "s3://asf-heat-pump-suitability/heat_network_desnz_data/heat-network-zone-map-Liverpool.gpkg"
+LSOA_JSON_PATH = os.path.join(INPUT_DIR, "liverpool_hp_suitability_lsoas.json")
+OUTPUT_DIR = os.path.join(
+    PROJECT_DIR, "asf_heat_pump_suitability/analysis/hn_zones/output_plots/"
+)
+
+# Ensure the output directory exists
+os.makedirs(OUTPUT_DIR, exist_ok=True)
+
+
+def load_data():
+    """
+    Load the necessary data files.
+
+    Returns:
+        Tuple containing:
+            - pd.DataFrame: DataFrame with Liverpool heat pump suitability scores.
+            - List[str]: List of LSOA codes for Liverpool.
+    """
+    try:
+        liverpool_hp_suitability_scores_pd = pd.read_parquet(
+            LIVERPOOL_HP_SUITABILITY_PARQUET
+        )
+        with open(LSOA_JSON_PATH, "r") as file:
+            liverpool_hp_suitability_lsoas = json.load(file)
+        return liverpool_hp_suitability_scores_pd, liverpool_hp_suitability_lsoas
+    except Exception as e:
+        logging.error(f"Error loading data: {e}")
+        raise
+
+
+def preprocess_data(
+    liverpool_hp_suitability_lsoas: list, lsoa_shp_path: str = LSOA_SHP_PATH
+) -> gpd.GeoDataFrame:
+    """
+    Preprocess the data by extracting unique LSOA geometries and filtering Liverpool LSOAs.
+
+    Args:
+        liverpool_hp_suitability_lsoas (list): List of LSOA codes for Liverpool.
+        lsoa_shp_path (str): Path to the LSOA shapefile.
+
+    Returns:
+        gpd.GeoDataFrame: GeoDataFrame with filtered Liverpool LSOA geometries.
+    """
+    lsoa_geometries_gdf = gpd.read_file(lsoa_shp_path)
+    if lsoa_geometries_gdf.crs != "EPSG:27700":  # Assuming British National Grid
+        logging.info("Converting to EPSG:27700")
+        lsoa_geometries_gdf = lsoa_geometries_gdf.to_crs("EPSG:27700")
+    liverpool_lsoa_geometries_gdf = lsoa_geometries_gdf[
+        lsoa_geometries_gdf["LSOA21CD"].isin(liverpool_hp_suitability_lsoas)
+    ]
+    return liverpool_lsoa_geometries_gdf
+
+
+def plot_lsoa_geometries(liverpool_lsoa_geometries_gdf, output_dir: str = OUTPUT_DIR):
+    """
+    Plot Liverpool LSOA geometries and save as PNG.
+
+    Args:
+        liverpool_lsoa_geometries_gdf (gpd.GeoDataFrame): GeoDataFrame with Liverpool LSOA geometries.
+        output_dir (str): Output directory to save the plot image.
+    """
+    fig, ax = plt.subplots(figsize=(12, 6))
+    liverpool_lsoa_geometries_gdf.plot(ax=ax)
+    plt.title("Liverpool LSOA Geometries")
+    plt.axis("off")
+    plt.savefig(os.path.join(output_dir, "liverpool_data_lsoas.png"))
+    plt.show()
+
+
+def merge_data(
+    liverpool_hp_suitability_scores_pd: pd.DataFrame,
```

Comment:
I think at least initially I like keeping the names descriptive especially on first iteration as I have a tendency to forget what each variable means! I'm also sure we can factor this down, as it will be for every LA rather than just Liverpool. But thinks like heat network (can abbreviate to hn as I'm very familiar with that by now).